app/subscriber/src/features/my-reports/MyReports.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { toast } from 'react-toastify';
+import { Tooltip } from 'react-tooltip';
+import { useReports } from 'store/hooks/subscriber/useReports';
+import { Col, FlexboxTable, IReportModel, Row } from 'tno-core';
+
+import { availableReportColumns } from './constants/availableReportColumns';
+import { columns } from './constants/columns';
+import * as styled from './styled';
+
+export const MyReport: React.FC = () => {
+  const [{ findAllReports, findMyReports, updateReport }] = useReports();
+  const [myReports, setMyReports] = React.useState<IReportModel[]>([]);
+  const [allReports, setAllReports] = React.useState<IReportModel[]>([]);
+
+  const [active, setActive] = React.useState<IReportModel>();
+  const [editable, setEditable] = React.useState<string>('');
+  React.useEffect(() => {
+    findMyReports().then((data) => {
+      setMyReports(data);
+    });
+    findAllReports().then((data) => {
+      setAllReports(data);
+    });
+    // Only do this on init.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const handleSave = () => {
+    if (!!active) {
+      updateReport(active).then((data) => {
+        toast.success(`${data.name} updated successfully`);
+        setMyReports([...myReports.filter((report) => report.id !== data.id), data]);
+        setEditable('');
+      });
```

Comment:
If you don't include a catch it will throw an error in the console for not handling a promise.  We have global handling but we pass the error along just in case the specific use case needs to do further logic.